Remove routes without any service_ids that cover today's date

EddyIonescu opened the pull request:
Fixes #581. Fixes #584.

<!-- Delete any of these headings if they aren't needed or applicable -->

## Proposed changes

Muni's GTFS contains each route twice, once from before February 22, and again starting from February 22, when they modified service on a few routes. This PR modifies save_routes to remove routes that don't have any service_ids that cover the date in which the script is being run, which is set to the current date (when versioned GTFS is added the date will be made an argument).

## Testing

Removes the duplicate routes, keeping the ones valid after February 22. Running Muni is still blocked by #576, which would need to be merged beforehand. 

- For TriMet, after removing the calendar.txt file and setting the date to a Sunday, all routes including the 92 South Beaverton Express (weekday peak-periods only) appear.
- For Muni, setting the date to February 22 (first date of changes) results in only the new routes being shown.

youngj commented on backend/save_routes.py:
```diff
@@ -50,10 +51,11 @@
     agencies = [config.get_agency(args.agency)] if args.agency is not None else config.agencies
 
     save_to_s3 = args.s3
+    date = date.today()
```
This variable shadows the class named `date`. To avoid this we have been using the variable name `d` for datetime.date instances.

youngj commented on backend/models/gtfs.py:
```diff
@@ -967,13 +976,44 @@ def get_route_data(self, route):
+        after the given date."""
+        trips_df = self.get_gtfs_trips()
+        dates_map = self.get_services_by_date(ignore_day_of_week=True)
+        active_services = set(dates_map[date])
```
Trimet's GTFS file doesn't have a calendar.txt to specify date ranges, only a calendar_dates.txt listing all dates when the service is active. It seems like if save_routes.py was run on a weekend or holiday where there were no service IDs associated with a particular route, this would say the route is not active. Perhaps instead it could check if today is between the first and last date for a service ID?

EddyIonescu replied:
Thanks for noticing this, didn't know that calendar.txt was optional. Also TriMet's GTFS does have calendar.txt now, I guess they added it recently (although their calendar_dates.txt still includes every date). Your approach sounds good - I've implemented it and tested it after deleting TriMet's calendar.txt file.

EddyIonescu commented:
@youngj Hey, thanks for catching that, sorry for making you do another review cycle. It looks like I forgot to rename `calendar.txt` even though I said I did, so that's why the 92 showed up on a Sunday. I've tested it again for March 8, 2020 with today's GTFS and with `calendar.txt` renamed and the 42 and 78 didn't appear, while the 92 and the 1 did. The 78a did appear, although looking in `calendar_dates.txt` it's supposed to.
I've also added a comment and renamed `_d` to something harder to miss.
Also let me know if you can run the job from February 22 - if not I can run it locally for Muni for the dates between then and now.
